tensornetwork/contractors/custom_path_solvers/nconinterface.py

```diff
@@ -0,0 +1,227 @@
+  """
+  Solve for the contraction order of a tensor network (encoded in the `ncon` 
+  syntax) that minimizes the computational cost. 
+  Args:
+    tensors: list of the tensors in the network.
+    connects: list of the tensor connections (in standard `ncon` format).
+    max_branch: maximum number of contraction paths to search at each step. 
+  Returns:
+    np.ndarray: the cheapest contraction order found (in ncon format).
+    float: the cost of the network contraction, given as log10(total_FLOPS).
+    bool: specifies if contraction order is guaranteed optimal.
+  """
+  # build log-adjacency matrix
+  log_adj = ncon_to_adj(tensors,connects)
+  
+  # run search algorithm
+  order, costs, is_optimal = full_solve_complete(log_adj,max_branch=max_branch)
+  
+  # put contraction order back into ncon format
+  cont_order = ord_to_ncon(connects,order)
+  
+  return cont_order, costs, is_optimal
+
+#------------------------------------------------------------------------#
+def ncon_to_adj(tensors: List[np.ndarray],
+                connects: List[List[int]]):
+  """
+  Create a log-adjacency matrix for a network from the `ncon` syntax. 
+  Args:
+    tensors: list of the tensors in the network.
+    connects: list of the tensor connections (in standard `ncon` format).
+  Returns:
+    np.ndarray: the log-adjacency matrix.
+  """
+  # process inputs  
+  N = len(connects)
+  ranks = [len(connects[i]) for i in range(N)]
+  flat_connects = np.hstack([connects[i] for i in range(N)])
+  tensor_counter = np.hstack([i*np.ones(ranks[i],dtype=int) for i in range(N)])
+  index_counter = np.hstack([np.arange(ranks[i]) for i in range(N)])
+  
+  # build log-adjacency index-by-index  
+  log_adj = np.zeros([N,N]);
+  unique_connects = np.unique(flat_connects)
+  for i in range(len(unique_connects)):
+      # identify tensor/index location of each edge
+      tnr = tensor_counter[flat_connects == unique_connects[i]]
+      ind = index_counter[flat_connects == unique_connects[i]]
+      if len(ind) == 1: # external index
+        log_adj[tnr[0],tnr[0]] += np.log10(tensors[tnr[0]].shape[ind[0]])
+      elif len(ind) == 2: # internal index
+        if (tnr[0] != tnr[1]): # ignore partial traces
+          log_adj[tnr[0],tnr[1]] += np.log10(tensors[tnr[0]].shape[ind[0]])
+          log_adj[tnr[1],tnr[0]] += np.log10(tensors[tnr[0]].shape[ind[0]])
+   
+  return log_adj
+
+#------------------------------------------------------------------------#
+def ord_to_ncon(connects: List[List[int]],
+                orders: np.ndarray):
+  """
+  Produces a `ncon` compatible index contraction order from the sequence of 
+  pairwise contractions. 
+  Args:
+    connects: list of the tensor connections (in standard `ncon` format).
+    orders: array of dim (2,N-1) specifying the set of N-1 pairwise 
+      tensor contractions.
+  Returns:
+    np.ndarray: the contraction order (in `ncon` format).
+  """
+  
+  N = 1 + (orders.size//2)
+  orders = orders.reshape(2,N-1)
```

Comment:
why are you doing this?

Reply:
For the `N=2` case it makes sense to accept `order` as a shape (2) array rather than a shape (2,1) array; the above code allows for both possibilities (although in hindsight checking `orders.ndim` would be more sensible)

Reply:
NP, just curious